stdlib/public/core/Character.swift

```diff
@@ -115,19 +116,40 @@ public struct Character :
       nativeString.append(s)
       _representation = .large(nativeString._core.nativeBuffer!._storage)
     }
+#else
+    // On big-endian systems (e.g. s390x), disable the optimization for now.
```

Comment:
This is a major performance hit for the common case.  Please leave a `FIXME(performance)` marker.